src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
                 {
-                    assm = Assembly.Load(new AssemblyName(assemblyName));
+                    try
```

Comment:
> At this point we are first trying to resolve it by assemblyname (string) and if that fails by AssemblyName type. So vice versa.

That wasn't 100% correct. First it tries to load it with the given assemblyname (string) and if that fails it create an AssemblyName type and tries to load it with the name property of it. I don't this is the same as ResolveSimpleAssemblyName.